src/gen_rules.ml

```diff
@@ -963,8 +998,63 @@ module Gen(P : Install_rules.Params) = struct
       Install_rules.Gen(struct
         include P
         let module_names_of_lib = module_names_of_lib
+        let mlds_of_dir = mlds_of_dir
       end) in
-    Install_rules.init ()
+    Install_rules.init ();
+    let docs_by_package =
+      let map = lazy (
+        SC.stanzas sctx
+        |> List.concat_map ~f:(fun (w : SC.Dir_with_jbuild.t) ->
+          List.filter_map w.stanzas ~f:(function
+            | Jbuild.Stanza.Documentation (d : Jbuild.Documentation.t) ->
+              Some (d.package.name, (w.ctx_dir, d))
+            | _ ->
+              None
+          ))
+        |> Package.Name.Map.of_list_multi
+      ) in
+      fun (p : Package.t) ->
+        Option.value (Package.Name.Map.find (Lazy.force map) p.name) ~default:[]
+    in
+    let modules_by_lib =
+      let lib_to_library = lazy (
+        SC.stanzas sctx
+        |> List.concat_map ~f:(fun (w : SC.Dir_with_jbuild.t) ->
+          List.filter_map w.stanzas ~f:(function
+            | Jbuild.Stanza.Library (l : Library.t) ->
+              Some (Library.best_name l, (w.ctx_dir, l))
+            | _ ->
+              None
+          ))
+        |> String_map.of_list_exn
```

Comment:
This might fail right? If we have several private libraries with the same name in different scopes

Reply:
Yeah, that's a bit unfortunate. I'll fix this, but maybe it makes sense to be able to retrieve the original `Jbuild.Library.t` from the lib's info?